### PR TITLE
Fix documentation comment for evmc_result::gas_left

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: include/evmc/evmc.h [hunk 1]
```diff
@@ -345,8 +345,8 @@ struct evmc_result
     /**
      * The amount of gas left after the execution.
      *
-     *  If evmc_result::code is not ::EVMC_SUCCESS nor ::EVMC_REVERT
-     *  the value MUST be 0.
+     * If evmc_result::status_code is not ::EVMC_SUCCESS nor ::EVMC_REVERT
```
Comment (gumb0) [inline include/evmc/evmc.h]:
```suggestion
     * If evmc_result::status_code is neither ::EVMC_SUCCESS nor ::EVMC_REVERT
```

Comment (gumb0) [reply include/evmc/evmc.h]: Why do you write status codes with double colon?

Comment (chfast) [reply include/evmc/evmc.h]: Doxygen will make them links.

Comment (chfast) [reply include/evmc/evmc.h]: https://evmc.ethereum.org/structevmc__result.html#af8478c93dbcc3cb2876037c5a5afd4c0

**Pull request merged**